plugins/mac-capture/mac-screen-capture.m

```diff
@@ -369,20 +358,13 @@ static bool init_screen_stream(struct screen_capture *sc)
         } break;
         case ScreenCaptureApplicationStream: {
             SCDisplay *target_display = get_target_display();
-            __block SCRunningApplication *target_application = nil;
-            {
-                [sc->shareable_content.applications
-                    indexOfObjectPassingTest:^BOOL(SCRunningApplication *_Nonnull application, NSUInteger idx,
-                                                   BOOL *_Nonnull stop) {
-                        if ([application.bundleIdentifier isEqualToString:sc->application_id]) {
-                            target_application = sc->shareable_content.applications[idx];
-                            *stop = TRUE;
-                        }
-                        return *stop;
-                    }];
+            NSMutableArray *target_application_array = [[NSMutableArray alloc] init];
```

Comment:
Given that the `target_application_array` will always contain a single element, I find it wasteful to use a mutable type for this.

Instead I'd rather declare an `NSArray` and then initialise it with the matching application object once it's found.

Reply:
Updated!